### DISCUSSION TITLE
Combobox active item should loop

### DISCUSSION THREAD

**Discussion by jakobvase** [tailwindlabs/headlessui#2999] [General]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

v1.7.18

**What browser are you using?**

Firefox

**Reproduction URL**

https://codesandbox.io/p/sandbox/staging-worker-hdkwqp

**Describe your issue**

I love your work! Thank you for tailwind and headlessui.

When the last item in a combobox is active, and I press `down`, I expect the first item in the combobox to be selected - same with the first item and pressing `up`. I expect the items to loop.

In the provided sandbox you can see the issue. You can also see it in your documentation here: https://headlessui.com/react/combobox.

**Comment by RobinMalfait** [comment]
Hey!

This is not a bug, but it is implemented this way deliberately. It allows you to hold down arrow up or arrow down and it will stop at the end or at the top respectively.

We have a version where it will loop around but _only_ if you are not holding down the arrow key.

If you want to jump from the last item to the top you can use the `Home` key, if you want to jump from the first item to the bottom then you can use the `End` key.

**Reply by jakobvase** [reply]
Hi Robin, thanks for the reply!

> We have a version where it will loop around but only if you are not holding down the arrow key.

This is what I'm looking for. Where is that version? How do I get it?

**Reply by Moise-VK** [reply]
I would also like to know which version this is

**Reply by pieterclaerhout** [reply]
Is that this one?

https://github.com/tailwindlabs/headlessui/tree/feat/fancy-wrap-around

**Comment by yinkakun** [comment]
Is there any workaround for this behaviour?
